parsl/usage_tracking/usage.py

```diff
@@ -110,28 +110,60 @@ def __init__(self, dfk, port=50077,
         self.python_version = "{}.{}.{}".format(sys.version_info.major,
                                                 sys.version_info.minor,
                                                 sys.version_info.micro)
-        self.tracking_enabled = self.check_tracking_enabled()
-        logger.debug("Tracking status: {}".format(self.tracking_enabled))
+        self.tracking_level = self.check_tracking_level()
+        self.start_time = None
+        logger.debug("Tracking level: {}".format(self.tracking_level))
+
+    def check_tracking_level(self) -> int:
+        """Check if tracking is enabled and return level.
+
+        Checks the following:
+            1. PARSL_TRACKING environment variable
+                - Possible values:
+                    ["true", "false", "True", "False", "0", "1", "2", "3"]
+                - Other values are treated as Level 0 (disabled)
+
+            2. usage_tracking in Config
+                - Possible values: [True, False, 0, 1, 2, 3]
+
+            Parsl will choose the lowest level that is specified by any of the parameters, and 0 if none of them are set
+
+            True/False values are treated as Level 1/Level 0 respectively.
+
+        Returns: int
+            - 0 : Tracking is disabled
+            - 1 : Tracking is enabled with level 1
+                  Share info about Parsl version, Python version, platform
+            - 2 : Tracking is enabled with level 2
+                  Share info about config + level 1
+            - 3 : Tracking is enabled with level 3
+                  Share info about app count, app fails, execution time + level 2
+        """
+        inf = sys.maxsize
 
-    def check_tracking_enabled(self):
-        """Check if tracking is enabled.
+        envvar = str(os.environ.get("PARSL_TRACKING", 0)).lower()
```

Comment:
does this default to 0, and then force tracking off if PARSL_TRACKING is not set?

Reply:
Changed this to default to None and setting `envvar_level = inf` in that case